File: Demo/Demo/Shapes/Shape.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Demo
{
	abstract class Shape : IShape
	{
		public Color Color { get; set; }
		public abstract Point Center { get; }

		public virtual void Print(TextWriter output)
		{
			output.WriteLine($"{this.Color} shape");
		}
	}
}

File: Demo/DemoVisualBasic/Shapes/Shape.vb

Imports System.Collections.Generic
Imports System.IO
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace Demo.Shapes
	MustInherit Class Shape
		Implements IShape
		Public Property Color As Color Implements IShape.Color

		Public MustOverride ReadOnly Property Center As Point Implements IShape.Center

		Public Overridable Sub Print(output As TextWriter) Implements IShape.Print
			output.WriteLine($"{Me.Color} shape")
		End Sub
	End Class
End Namespace
